src/intrinsics/prepack/global.js

```diff
@@ -264,13 +264,29 @@ export default function(realm: Realm): void {
         // casting to any to avoid Flow bug "*** Recursion limit exceeded ***"
         if ((object: any) instanceof AbstractObjectValue || (object: any) instanceof ObjectValue) {
           let generator = realm.generator;
-          if (generator)
-            generator.emitInvariant(
-              [object, value, object],
-              ([objectNode, valueNode]) =>
-                t.binaryExpression("!==", t.memberExpression(objectNode, t.identifier(key)), valueNode),
-              objnode => t.memberExpression(objnode, t.identifier(key))
-            );
+          if (generator) {
+            let condition = ([objectNode, valueNode]) =>
+              t.binaryExpression("!==", t.memberExpression(objectNode, t.identifier(key)), valueNode);
+            if (invariantOptions && invariantOptions instanceof StringValue) {
+              switch (invariantOptions.value) {
+                case "DEFINED":
```

Comment:
The point of this PR is that we should be able skip or reduce the strictness of the invariant generated (because checking `===` on FunctionValues will always fail as currently written). This is so that we can at least check that a value exists instead of eliding the check entirely.